lib/services/service_brokers/v2/catalog_schemas.rb

```diff
@@ -13,26 +16,94 @@ def valid?
 
     private
 
-    def validate_and_populate_create_instance(attrs)
-      return unless attrs
-      unless attrs.is_a? Hash
-        errors.add("Schemas must be a hash, but has value #{attrs.inspect}")
+    def validate_and_populate_create_instance(schemas)
+      return unless schemas
+      unless schemas.is_a? Hash
+        errors.add("Schemas must be a hash, but has value #{schemas.inspect}")
         return
       end
 
       path = []
       ['service_instance', 'create', 'parameters'].each do |key|
         path += [key]
-        attrs = attrs[key]
-        return nil unless attrs
+        schemas = schemas[key]
+        return nil unless schemas
 
-        unless attrs.is_a? Hash
-          errors.add("Schemas #{path.join('.')} must be a hash, but has value #{attrs.inspect}")
+        unless schemas.is_a? Hash
+          errors.add("Schemas #{path.join('.')} must be a hash, but has value #{schemas.inspect}")
           return nil
         end
       end
 
-      @create_instance = attrs
+      create_instance_schema = schemas
+      create_instance_path = path.join('.')
+
+      validate_schema(create_instance_path, create_instance_schema)
+      return unless errors.empty?
+
+      @create_instance = create_instance_schema
```

Comment:
(This comment doesn't necessarily reflect changes form this specific PR, but the set of PRs)

We were also surprised to see the object being created conditionally through the validations. We'd prefer that the object's validity rely on `valid?`'s return value, rather than presence of an attribute.